File: src/main/java/co/selim/exiftool4j/ExifKey.java
package co.selim.exiftool4j;

import java.util.Arrays;

/**
 * An enum of the exif key value pairs that will be extracted.
 */
public enum ExifKey {
    DESCRIPTION("ImageDescription"),
    DATE_TIME("DateTimeOriginal"),
    DEVICE_MAKE("Make"),
    DEVICE_MODEL("Model"),
    LENS("LensModel"),
    APERTURE("Aperture"),
    FOCAL_LENGTH("FocalLength"),
    SHUTTER_SPEED("ShutterSpeed"),
    ISO("ISO"),
    FLASH_STATUS("Flash"),
    WIDTH("ImageWidth"),
    HEIGHT("ImageHeight");

    private String exifToolName;

    ExifKey(String exifToolName) {
        this.exifToolName = exifToolName;
    }

    ExifKey fromExifToolString(String exifToolName) {
        return Arrays.stream(values())
                .filter(exifKey -> exifToolName.equals(exifKey.exifToolName))
                .findFirst()
                .orElseThrow(() -> new IllegalStateException("Exif key not found for name " + exifToolName));
    }

    String getExifToolName() {
        return exifToolName;
    }
}

File: src/main/java/co/selim/exiftool4j/ExifKey.kt
package co.selim.exiftool4j

enum class ExifKey(internal val names: List<String>, internal val exclude: List<String> = emptyList()) {
    DATE_TIME(listOf("Date/Time Original", "Create Date", "Digital Creation Date/Time")),
    MAKE(listOf("Make")),
    MODEL(listOf("Model", "Camera Model Name")),
    LENS(listOf("Lens", "Lens Model", "Lens Type", "Lens ID")),
    APERTURE(listOf("Aperture")),
    SHUTTER_SPEED(listOf("Shutter Speed", "Exposure Time")),
    ISO(listOf("ISO", "Program ISO")),
    FLASH_STATUS(listOf("Flash")),

    // exclude equivalent focal lengths
    FOCAL_LENGTH(listOf("Focal Length"), exclude = listOf("equivalent"))
}